service_container/injection_types.rst

```diff
@@ -100,6 +100,103 @@ working with optional dependencies. It is also more difficult to use in
 combination with class hierarchies: if a class uses constructor injection
 then extending it and overriding the constructor becomes problematic.
 
+Immutable-setter Injection
+--------------------------
+
+Another possible injection in to use a method which return ``static``,
```

Comment:
which returns a separate instance by cloning then configuring the original service

Reply:
s/in to use/is to use/
s/which return/which returns

Reply:
Improved, to check :)